### DISCUSSION TITLE
PMU Firmware

### DISCUSSION THREAD

**Discussion by edisionnano** [NVIDIA/open-gpu-kernel-modules#277] [Q&A]
While it is stated that this driver strictly pertains to Ampere, Turing and upcoming architectures, Maxwell 2 and Pascal which are very popular and capable GPU architectures (especially the latter) while still supported by the propriatary kernel driver no open source kernel driver can truly unlock their full potential since the cards run at their lowest clocks and PMU firmware is necessary in order for nouveau to reclock them. While nvidia doesn't seem to have plans to extend this driver to support these generations it would be nice if they provided PMU firmware to allow nouveau or any other effort to use these cards to the max of their abilities by providing PMU firmware and any other firmware necessary (for thermals, etc.) as well as some documentation.

**Comment by YusufKhan-gamedev** [comment]
Maybe if someone spent the time to remove the dependence on GSP would a Open Source graphics stack be possible...food for thought.

**Reply by edisionnano** [reply]
No, we need PMU firmware officially released by nvidia with a license that allows nouveau to use it, signed ofc

**Reply by YusufKhan-gamedev** [reply]
Is there any reason why adding ogk support would be counterproductive to anything?

**Reply by edisionnano** [reply]
What is ogk?

**Reply by YusufKhan-gamedev** [reply]
[open-gpu-kernel-modules](https://github.com/NVIDIA/open-gpu-kernel-modules)

**Reply by edisionnano** [reply]
Wouldn't that be ogkm?

**Reply by YusufKhan-gamedev** [reply]
Not as catchy, say them outloud ogkm sounds eww

**Reply by Evernow** [reply]
> Is there any reason why adding ogk support would be counterproductive to anything?

Because upstream refuses to allow NVIDIA's kernel driver even though it works fine (just look at SUSE, one of the largest enterprise distros defaulting to it already after just a month of its first release), so I doubt Nouveau will support it. If you're going to blame someone blame upstream for their perfectionist policies.

**Reply by YusufKhan-gamedev** [reply]
@Evernow I doubt upstream anything is to blame, there is no free userspace yet so they dont care to adopt it, nvk should have at least basic support for it soon after upstreaming...but id guess that ogk would consume the same realm as kgsl does in freedreno....

**Reply by yoshimo** [reply]
Where is it for a 980? So far all we hear/read is that it is still missing.

**Reply by YusufKhan-gamedev** [reply]
> > we need PMU firmware officially released by nvidia with a license that allows nouveau to use it, signed ofc
> 
> Signed PMU, you mean? But it was released. You do not need any license, it is nonfree. Since when nonfree is a problem for Debian, e.g., or some of its forks, like Kali or Ubuntu? Even though you cannot destribute binaries, as this will violate GPL, end users are allowed to compile code thenselves and use binary bolbs, this either does not violate GPL or it is not prosecutable.

Not released, prohibitively hard to extract from the blob.

**Comment by ghost** [comment]


**Reply by edisionnano** [reply]
I doubt that's a solution. These cards need firmware in order to reclock, They have the Falcon chip.

**Reply by YusufKhan-gamedev** [reply]
Probably....one way to find out!

**Comment by edisionnano** [comment]
Bump.
PMU firmware is very important, other types of firmware have been provided to linux firmware for pascal, volta and maxwell time for PMU too

**Comment by zhjihuang** [comment]
Bumping this in 2026 after official support for these cards have ended.

**Reply by atrtr23** [reply]
+1 bump for my Maxwell and Pascal brethrens

**Comment by Steperman5692** [comment]
NVIDIA wouldn't lose much from releasing the PMU firmware for Maxwell/Pascal/Volta as they GPUs are basically EoL with quarter security updates till Oct 2028.
Letting the community support these GPUs after Oct 2028 would help in the long run as they're still people who still uses these GPUs & probably don't have the money to buy another one. 
As the NVIDIA Linux world is moving to towards Nova/OKM, The PMU firmware would be the only critical component that `nouveau` needs for breathing new life to these GPUs if NVIDIA no longer wants to support them. They can simply release the proprietary PMU firmware blob in `linux-firmware` just like they did with the GSP firmware & the Tegra X1.

**Reply by AIVirtuoso** [reply]
I agree, it wouldn't cost anything to do
